Crashes in initial scan

The walker has a nil event logger;

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x46fbe67]

goroutine 167 [running]:
github.com/syncthing/syncthing/lib/scanner.(*walker).walk.func2.1(0xc0106df3a0, 0xc010a06ae0, 0xc000b220a0, 0xc000467db0, 0x19b13284, 0x4c5c840, 0xc000448200)
        /Users/jb/dev/syncthing/syncthing/lib/scanner/walk.go:173 +0x77
created by github.com/syncthing/syncthing/lib/scanner.(*walker).walk.func2
        /Users/jb/dev/syncthing/syncthing/lib/scanner/walk.go:160 +0x334
```

--- a/lib/model/folder.go
+++ b/lib/model/folder.go
@@ -348,6 +348,7 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 		ProgressTickIntervalS: f.ScanProgressIntervalS,
 		LocalFlags:            f.localFlags,
 		ModTimeWindow:         f.ModTimeWindow(),
+		EventLogger:           f.evLogger,
 	})
 
 	batchFn := func(fs []protocol.FileInfo) error {

--- a/lib/scanner/walk.go
+++ b/lib/scanner/walk.go
@@ -57,7 +57,7 @@ type Config struct {
 	// Modification time is to be considered unchanged if the difference is lower.
 	ModTimeWindow time.Duration
 	// Event logger to which the scan progress events are sent
-	EvLogger events.Logger
+	EventLogger events.Logger
 }
 
 type CurrentFiler interface {
@@ -170,7 +170,7 @@ func (w *walker) walk(ctx context.Context) chan ScanResult {
 					current := progress.Total()
 					rate := progress.Rate()
 					l.Debugf("Walk %s %s current progress %d/%d at %.01f MiB/s (%d%%)", w.Folder, w.Subs, current, total, rate/1024/1024, current*100/total)
-					w.EvLogger.Log(events.FolderScanProgress, map[string]interface{}{
+					w.EventLogger.Log(events.FolderScanProgress, map[string]interface{}{
 						"folder":  w.Folder,
 						"current": current,
 						"total":   total,

--- a/lib/scanner/walk_test.go
+++ b/lib/scanner/walk_test.go
@@ -788,8 +788,8 @@ func testConfig() Config {
 	evLogger := events.NewLogger()
 	go evLogger.Serve()
 	return Config{
-		Filesystem: testFs,
-		Hashers:    2,
-		EvLogger:   evLogger,
+		Filesystem:  testFs,
+		Hashers:     2,
+		EventLogger: evLogger,
 	}
 }

